### PR TITLE
Adding AFD Service Tag Check policy for consumption tier

### PR DESCRIPTION
**Pull request by WillEastbury**

### CODE REVIEW

File: examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            <value>{{nvfdid}}</value>
+        </check-header>
+        <validate-jwt header-name="Authorization" failed-validation-httpcode="401" failed-validation-error-message="No token provided or token is invalid." require-expiration-time="true" require-signed-tokens="true" clock-skew="300">
+            <openid-config url="{{nvoidc}}" />
+            <audiences>
+                <audience>{{nvjwtaud}}</audience>
+            </audiences>
+        </validate-jwt>
+        <set-header name="Authorization" exists-action="delete" />
+        <cache-lookup-value key="cacheresponse1" variable-name="cacheresponse1" />
```
Comment (VitaliyKurokhtin) [inline examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml:21]: Why "1"?

File: examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    return Newtonsoft.Json.JsonConvert.SerializeObject(rootvalues.Children().Where(e => (string)e["id"] == "AzureFrontDoor.Backend").FirstOrDefault()["properties"]["addressPrefixes"].Select(c => (string)c).ToList());    
+                }" />
+            </when>
+        </choose>
+        <choose>
+            <when condition="@{       
+
+                    int HostToNetworkOrder(int host)
+                    {
+                        return (((int)HostToNetworkOrderShort((short)host) & 0xFFFF) << 16) | ((int)HostToNetworkOrderShort((short)(host >> 16)) & 0xFFFF);
```
Comment (VitaliyKurokhtin) [inline examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml:41]: Shouldn't bytes within a word also be rotated?

File: examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                <!-- If the arm context variable doesn’t exist, make an HTTP request to retrieve it.  -->
+                <send-request mode="new" timeout="20" ignore-error="false" response-variable-name="cacheresponse1">
+                    <set-url>https://management.azure.com/subscriptions/{{nvazsid}}/providers/Microsoft.Network/locations/eastus/serviceTags?api-version=2021-03-01</set-url>
+                    <set-method>GET</set-method>
+                    <authentication-managed-identity resource="https://management.azure.com" />
+                </send-request>
+                <set-variable name="cacheresponse1" value="@{
+                    JArray rootvalues = (JArray)((IResponse)context.Variables["cacheresponse1"]).Body.As<JObject>()["values"];
+                    return Newtonsoft.Json.JsonConvert.SerializeObject(rootvalues.Children().Where(e => (string)e["id"] == "AzureFrontDoor.Backend").FirstOrDefault()["properties"]["addressPrefixes"].Select(c => (string)c).ToList());    
+                }" />
```
Comment (VitaliyKurokhtin) [inline examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml:33]: List of strings would be more efficiently stored in a string with a separator, i.e. string1|string2|string3

File: examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                        string[] inputIPParts = ipAddress.Split('.');
+                        string[] cidrIPArray = cidrParts[0].Split('.');
+
+                        if (inputIPParts.Length == 4 && cidrIPArray.Length == 4)
+                        {
+                            byte[] inputIPBytes = new byte[] {Convert.ToByte(int.Parse(inputIPParts[0])), Convert.ToByte(int.Parse(inputIPParts[1])), Convert.ToByte(int.Parse(inputIPParts[2])), Convert.ToByte(int.Parse(inputIPParts[3]))};
+                            byte[] cidrIPBytes = new byte[] {Convert.ToByte(int.Parse(cidrIPArray[0])), Convert.ToByte(int.Parse(cidrIPArray[1])), Convert.ToByte(int.Parse(cidrIPArray[2])), Convert.ToByte(int.Parse(cidrIPArray[3]))};
+                    
+                            int cidrAddr = BitConverter.ToInt32(inputIPBytes,0);
+                            int ipAddr = BitConverter.ToInt32(cidrIPBytes,0);
```
Comment (VitaliyKurokhtin) [inline examples/CheckInboundIPIsInAFDServiceTagRangeOnConsumptionTier.policy.xml:66]: Doing bit shift operations would be more efficient

**Review by VitaliyKurokhtin — CHANGES_REQUESTED** [review]

**Comment by adrianhall** [comment]
Closing due to age and lack of response.

**Pull request closed without merging**